### DISCUSSION TITLE
Secrets are no more showed on the UI

### DISCUSSION THREAD

**Discussion by rafaarcangel** [skoruba/IdentityServer4.Admin#1058] [Q&A]
Hi all....

I realized that from the previous version 1.0.0 to the new 2.0.0, there are a few new lines to set ClientSecrets values to null.
This new code shows an empty box when you try to see those secrets in the UI
![image](https://github.com/skoruba/IdentityServer4.Admin/assets/19646006/d5096b83-7361-4e14-97f8-d77af719b6ad)

From code ClientService.cs =>
```
clientSecretsDto.ClientSecrets.ForEach(delegate (ClientSecretDto x)
{
    x.Value = null;
});
```

With this change, from now on... how is it possible to see a secret from UI?

**Comment by skoruba** [comment]
I removed it very long time ago, because it was not recommended to be able to see this value there. it is from security reason.

**Comment by rafaarcangel** [comment]
Ok, I understand and agree but, in that case, why keep showing the show secrets section? If we can't use it anymore (at least only create them), it makes noise for our clients, and it's annoying to show an empty box.

So, to preserve this functionality, we should make a detour (and probably override the logic) under our criteria and security risk.

Anyway, thanks a lot for your reply.
I'll share it with my client!
Regards

**Reply by skoruba** [reply]
This section is not there, maybe you have older version of UI?

**Reply by skoruba** [reply]
Please, consider also update to Duende IdentityServer, because IdentityServer4 is not supported. This repository will be archived soon - new repository with new Admin UI is here: https://github.com/skoruba/Duende.IdentityServer.Admin
